services/debugger/cache/internal/badger/badger.go

```diff
@@ -127,9 +130,11 @@ func New[E any](origin string, logger logger.Logger, opts ...func(Cache[E])) (*C
 		return nil, err
 	}
 	storagePath := path.Join(tmpDirPath, badgerPathName)
+	fmt.Println(storagePath)
 	e.path = storagePath
 	e.done = make(chan struct{})
 	e.closed = make(chan struct{})
+	e.stats = stats.NewStats(config.Default, logger.Default, svcMetric.Instance)
```

Comment:
Can we pass stats as a function parameter, similar to what we are doing for `logger.Logger`?